compile errors with a number of Ports on ubuntu 21.04 gcc-11 g++-11

I cannot compile curl, wget and many others get errors like this 
```
In file included from include/internal/sockets.h:83,
                 from crypto/bio/bio_local.h:11,
                 from crypto/bio/bss_dgram.c:13:
crypto/bio/bss_dgram.c: In function 'dgram_get_mtu_overhead':
crypto/bio/bss_dgram.c:373:20: error: invalid type argument of '->' (have 'struct in6_addr')
  373 |                 && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                    ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:373:20: error: invalid type argument of '->' (have 'struct in6_addr')
  373 |                 && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                    ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:373:20: error: invalid type argument of '->' (have 'struct in6_addr')
  373 |                 && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                    ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:373:20: error: invalid type argument of '->' (have 'struct in6_addr')
  373 |                 && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                    ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:373:20: error: invalid type argument of '->' (have 'struct in6_addr')
  373 |                 && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                    ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:373:20: error: invalid type argument of '->' (have 'struct in6_addr')
  373 |                 && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                    ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:373:20: error: invalid type argument of '->' (have 'struct in6_addr')
  373 |                 && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                    ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:373:20: error: invalid type argument of '->' (have 'struct in6_addr')
  373 |                 && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                    ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:373:20: error: invalid type argument of '->' (have 'struct in6_addr')
  373 |                 && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                    ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:373:20: error: invalid type argument of '->' (have 'struct in6_addr')
  373 |                 && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                    ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:373:20: error: invalid type argument of '->' (have 'struct in6_addr')
  373 |                 && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                    ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:373:20: error: invalid type argument of '->' (have 'struct in6_addr')
  373 |                 && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                    ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c: In function 'dgram_ctrl':
crypto/bio/bss_dgram.c:544:24: error: invalid type argument of '->' (have 'struct in6_addr')
  544 |                     && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                        ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:544:24: error: invalid type argument of '->' (have 'struct in6_addr')
  544 |                     && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                        ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:544:24: error: invalid type argument of '->' (have 'struct in6_addr')
  544 |                     && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                        ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:544:24: error: invalid type argument of '->' (have 'struct in6_addr')
  544 |                     && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                        ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:544:24: error: invalid type argument of '->' (have 'struct in6_addr')
  544 |                     && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                        ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:544:24: error: invalid type argument of '->' (have 'struct in6_addr')
  544 |                     && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                        ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:544:24: error: invalid type argument of '->' (have 'struct in6_addr')
  544 |                     && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                        ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:544:24: error: invalid type argument of '->' (have 'struct in6_addr')
  544 |                     && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                        ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:544:24: error: invalid type argument of '->' (have 'struct in6_addr')
  544 |                     && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                        ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:544:24: error: invalid type argument of '->' (have 'struct in6_addr')
  544 |                     && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                        ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:544:24: error: invalid type argument of '->' (have 'struct in6_addr')
  544 |                     && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                        ^~~~~~~~~~~~~~~~~~~~
crypto/bio/bss_dgram.c:544:24: error: invalid type argument of '->' (have 'struct in6_addr')
  544 |                     && IN6_IS_ADDR_V4MAPPED(&tmp_addr))
      |                        ^~~~~~~~~~~~~~~~~~~~
make[1]: *** [Makefile:1521: crypto/bio/bss_dgram.o] Error 1
make[1]: *** Waiting for unfinished jobs....
crypto/x509/x509_obj.c: In function 'X509_NAME_oneline':
crypto/x509/x509_obj.c:55:9: warning: 'strncpy' is deprecated: use strlcpy or String::copy_characters_to_buffer [-Wdeprecated-declarations]
   55 |         strncpy(buf, "NO X509_NAME", len);
      |         ^~~~~~~
In file included from include/internal/cryptlib.h:14,
                 from crypto/x509/x509_obj.c:11:
/home/grizzlysmit/Projects/sernityos/serenity/Build/i686/Root/usr/include/string.h:33:87: note: declared here
   33 | __attribute__((deprecated("use strlcpy or String::copy_characters_to_buffer"))) char* strncpy(char* dest, const char* src, size_t);
      |                                                                                       ^~~~~~~
make[1]: Leaving directory '/home/grizzlysmit/Projects/sernityos/serenity/Ports/openssl/openssl-1.1.1k'
make: *** [Makefile:174: all] Error 2
```
is it something I don't have installed.

--- a/Ports/openssl/package.sh
+++ b/Ports/openssl/package.sh
@@ -8,7 +8,7 @@ files="https://ftp.nluug.nl/security/openssl/openssl-${version}.tar.gz openssl-$
 auth_type=sha256
 
 depends="zlib"
-configopts="--prefix=/usr/local -DOPENSSL_SYS_SERENITY=1 gcc zlib no-tests no-threads no-asm"
+configopts="--prefix=/usr/local -DOPENSSL_SYS_SERENITY=1 -DOPENSSL_USE_IPV6=0 gcc zlib no-tests no-threads no-asm"
 
 configure() {
     run ./"$configscript" $configopts

